script/cache_critical_dns

```diff
@@ -213,11 +224,25 @@ class HealthyCache
     @cached = nil  # a single IP address that was most recently found to be healthy
   end
 
+  # Returns the first healthy server found in the list of resolved addresses.
+  # If the ResolverCache is empty, returns the currently cached healthy servers
+  # (if any).
```

Comment:
I would suggest a revision to the second sentence:

> Returns the last known healthy server if all servers disappear from the DNS.

In only a few words, this phrasing explains under what conditions the `ResolverCache` may be empty, and also why the cache is helpful.